src/main/java/Deck.java

```diff
@@ -1,30 +1,32 @@
-
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
 
 public class Deck {
 
-private final Card[] deck = new Card[52];
+    private final Card[] deck = new Card[52];
 
-public Deck() {
-    newDeck();
-}
+    public Deck() {
+        newDeck();
+    }
 
-public void newDeck(){
-    int ix = 0;
-    for(Card.Suit suit : Card.Suit.values()){
-        for(Card.Rank rank : Card.Rank.values()){
-            deck[ix++] = new Card(suit, rank);
+    public void newDeck(){
+        int ix = 0;
+        for(Card.Suit suit : Card.Suit.values()){
+            for(Card.Rank rank : Card.Rank.values()){
+                deck[ix++] = new Card(suit, rank);
+            }
         }
     }
-}
 
-public Card nextCard(){
-    Card card = null;
-    while(card == null){
-        int ix = (int) (Math.random() * deck.length);
-        card = deck[ix];
-        deck[ix] = null;
+    public Card nextCard(){
+        Card card = null;
+        while(card == null){
+            int ix = (int) (Math.random() * deck.length);
+            card = deck[ix];
+            deck[ix] = null;
+        }
+        return card;
```

Comment:
Once you get down to the last card in the deck, this method of randomization becomes very expensive. 
There are a lot better ways to shuffle the deck. 

Also, what happens when you have no more cards in the deck?